Source/WebCore/platform/network/curl/CurlRequest.cpp

```diff
@@ -748,6 +751,14 @@ bool CurlRequest::isHandlePaused() const
     return m_isHandlePaused;
 }
 
+bool CurlRequest::shouldAbortConnectRequest(const URL& url, long statusCode, long httpConnectCode)
+{
+    if (url.protocolIs("https"_s) && !statusCode && httpConnectCode && httpConnectCode != 200 && httpConnectCode != 407)
```

Comment:
Do you think "https" check is needed here? I guess httpConnectCode is zero in case of http. Right? If httpConnectCode is zero, this check fails as expected.

Reply:
> How does curl command handle the case? Does curl command just ignore the connect error? Is this a curl bug?

Good point! The curl handle those cases correctly. WebKit's curl port seems to be misimplemented.

```
> curl --version
curl 7.85.0 (x86_64-w64-mingw32) libcurl/7.85.0 OpenSSL/1.1.1q (Schannel) zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.3 libssh2/1.10.0 nghttp2/1.48.0
Release-Date: 2022-08-31
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL SSPI threadsafe TLS-SRP zstd

> curl https://webkit.org -x http://127.0.0.1:8888
curl: (56) Received HTTP code 302 from proxy after CONNECT
```

https://github.com/curl/curl/blob/curl-7_85_0/lib/http_proxy.c#L670-L694
```
  if(data->info.httpproxycode/100 != 2) {
    ...
    failf(data, "Received HTTP code %d from proxy after CONNECT",
          data->req.httpcode);
    return CURLE_RECV_ERROR;
  }
```

If the result of `CurlRequest::didCompleteTransfer` is not `CURLE_OK`, it seems that `invokeDidReceptionResponse` should not be executed.

https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/network/curl/CurlRequest.cpp#L469-L476
```
void CurlRequest::didCompleteTransfer(CURLcode result)
{
   ...

    if (needToInvokeDidReceiveResponse()) {
        // Processing of didReceiveResponse() has not been completed. (For example, HEAD method)
        // When completeDidReceiveResponse() is called, didCompleteTransfer() will be called again.

        m_finishedResultCode = result;
        invokeDidReceiveResponse(m_response, Action::FinishTransfer);  <<== HERE
        return;
    }
    ...
}
```